docs/intentReaderHandlers/ClearEventListeners.md

```diff
@@ -0,0 +1,61 @@
+# ClearEventListeners
+
+## Overview
+
+ClearEventListeners handler is invoked when the task specified in the intent has the value "clearAllListeners". This handler is used to deregister all the events that are registered already. 
+
+It performs the following actions :
+1. Deregister all the events.
+2. Save the response/error.
+3. Format the result and send the response back to IntentReader.
+
+## Usage
+* This handler is invoked to clear all the event listeners at a time.
+
+```json
+    {
+        "action": "search",
+        "data": {
+            "query": {
+                "task": "clearAllListeners",
+                "action": "NA",
+                "appType": "<appType>"
+            }
+        },
+        "context": {
+            "source": "device"
+        }
+    }
+```
+
+### Parameters
+
+| Key                   | Description                                                                           | Required? |
+|-----------------------|---------------------------------------------------------------------------------------|-----------|
+| task                  | "clearAllListeners"- Its a static value and should not be changed for this handler    | Y         |
+| appType               | Corresponding intent is launching on which app                                        | Y         |
+
+
+## Examples
```

Comment:
Is there no example for an invalid request of this type? If it is not really possible to incorrectly invoke this can we just add a note about why an invalid scenario is not applicable.

Reply:
### Invalid Intent and Response
Invalid response is already present

Reply:
Added comment as ->  An invalid scenario is not possible for this handler as it is not accepting parameters or additional information for deregistering all events.

Reply:
What happens if params are passed? Will it just ignore them? Can we add that info?